apps/meteor/ee/app/livechat-enterprise/server/api/lib/priorities.ts

```diff
@@ -46,6 +46,19 @@ export async function findPriorities({ text, pagination: { offset, count, sort }
 	};
 }
 
-export async function findPriorityById({ priorityId }: FindPrioritiesByIdParams): Promise<FindPrioritiesByIdResult> {
-	return OmnichannelServiceLevelAgreements.findOneById(priorityId);
+export async function findPriorityById({ priorityId }: FindPriorityByIdParams): Promise<FindPriorityByIdResult> {
+	return LivechatPriority.findOneById(priorityId);
+}
+
+export async function createPriority(data: ILivechatPriorityData): Promise<ILivechatPriority | false> {
+	const exists = await LivechatPriority.findOneByIdOrName(data.name);
+	if (exists) {
+		return false;
+	}
+	const insert = await LivechatPriority.insertOne(data);
+	if (!insert.acknowledged) {
+		return false;
+	}
+	Object.assign(data, { _id: insert.insertedId });
+	return data as ILivechatPriority;
```

Comment:
Can we replace this logic with something like findOneAndUpdate with $setOnInsert? 👀

Reply:
Other thing: it's ok to just throw an error directly from here. Something like `throw new Error('Priority already exists') or ('cannot create priority') should be enough 🤔 instead of return false.